File: bindings/csharp/Sop/Redis.cs
using System;
using System.Runtime.InteropServices;

namespace Sop;

/// <summary>
/// Provides methods to manage the global Redis connection.
/// Note: In newer versions of SOP, Redis configuration can be passed directly via DatabaseOptions,
/// allowing for different Redis connections per Database instance.
/// This static class is maintained for backward compatibility or global initialization.
/// </summary>
public static class Redis
{
    /// <summary>
    /// Initializes the global shared Redis connection.
    /// </summary>
    public static void Initialize(string url)
    {
        var resPtr = NativeMethods.OpenRedisConnection(Interop.ToBytes(url));
        var res = Interop.FromPtr(resPtr);
        if (res != null) throw new SopException(res);
    }

    /// <summary>
    /// Closes the global shared Redis connection.
    /// </summary>
    public static void Close()
    {
        var resPtr = NativeMethods.CloseRedisConnection();
        var res = Interop.FromPtr(resPtr);
        if (res != null) throw new SopException(res);
    }
}

File: bindings/java/src/main/java/com/sharedcode/sop/Redis.java
package com.sharedcode.sop;

import com.sun.jna.Pointer;

public class Redis {
    /**
     * Initializes the global shared Redis connection.
     * @param url The Redis connection string (e.g., "redis://localhost:6379")
     * @throws SopException if initialization fails
     */
    public static void initialize(String url) throws SopException {
        Pointer resPtr = SopLibrary.INSTANCE.openRedisConnection(url);
        String error = SopUtils.fromPointer(resPtr);
        if (error != null && !error.isEmpty()) {
            throw new SopException(error);
        }
    }

    /**
     * Closes the global shared Redis connection.
     * @throws SopException if closing fails
     */
    public static void close() throws SopException {
        Pointer resPtr = SopLibrary.INSTANCE.closeRedisConnection();
        String error = SopUtils.fromPointer(resPtr);
        if (error != null && !error.isEmpty()) {
            throw new SopException(error);
        }
    }
}
